modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -159,6 +172,28 @@ func shouldRedeploy(bd *fleet.BundleDeployment) bool {
 	return *bd.Status.SyncGeneration != bd.Spec.Options.ForceSyncGeneration
 }
 
+func (h *handler) cleanupOldAgent(bd *fleet.BundleDeployment, modified []fleet.ModifiedStatus) error {
+	if !isAgent(bd) {
+		return nil
+	}
```

Comment:
nit: We should considering adding a NOP trace log.

Reply:
logging a noop would be too much.  I moved the if out of the method to make it less weird.